version: '3.4'

services:
  patientsresolver.api:
    image: ${DOCKER_REGISTRY-}patientsresolverapi
    build:
      context: .
      dockerfile: PatientsResolver.API/Dockerfile
    volumes:
      - "C:/TestData:/data"

  influencecalculator.api:
    image: ${DOCKER_REGISTRY-}influencecalculatorapi
    build:
      context: .
      dockerfile: InfluenceCalculator.API/Dockerfile

  patientdatahandler.api:
    image: ${DOCKER_REGISTRY-}patientdatahandlerapi
    build:
      context: .
      dockerfile: PatientDataHandler.API/Dockerfile

  rabbitmq:
    container_name: rabbitmq
    tty: true
    hostname: rabbitmq
    ports:
        - "15672:15672"
        - "5672:5672"
    image: rabbitmq:3-management
    environment:
        - RABBITMQ_DEFAULT_USER=user
        - RABBITMQ_DEFAULT_PASS=password

  postgres_image:
    container_name: postgresql
    image: postgres:latest
    ports:
      - "5432"
    restart: always
    volumes:
      - db_volume:/var/lib/postgresql/data
      #- ./dbscripts/seed.sql :/docker-entrypoint-initdb.d/seed.sql
    environment:
      POSTGRES_USER: "postgres"
      POSTGRES_PASSWORD: "lisa2021"
      POSTGRES_DB: "PatientData_db"
    networks:
      - postgres

  pgadmin:
     container_name: pgadmin_container
     image: dpage/pgadmin4:5.7
     environment:
         PGADMIN_DEFAULT_EMAIL: "70dog@mail.ru"
         PGADMIN_DEFAULT_PASSWORD: "password"
         PGADMIN_CONFIG_SERVER_MODE: "False"
     volumes:
      - ./pgadmin:/var/lib/pgadmin
     ports:
      - "5050:80"
     restart: unless-stopped
     networks:
      - postgres

volumes:
   db_volume:
   pgadmin:

networks:
  postgres:
    driver: bridge

    

